### PR TITLE
Convert some PreCert stuff and Cert loading from DER to util::Status

### PR DESCRIPTION
**Pull request by Martin2112**
Still a few more bits to convert but we're getting near to the end of this now!

### CODE REVIEW

File: cpp/log/cert_test.cc [hunk 1]
```diff
@@ -306,7 +306,7 @@ TEST_F(CertTest, LoadValidFromDer) {
   string der;
   ASSERT_OK(leaf.DerEncoding(&der));
   Cert second;
-  EXPECT_EQ(Cert::TRUE, second.LoadFromDerString(der));
+  EXPECT_TRUE(second.LoadFromDerString(der).ok());
```
Comment (pphaneuf) [inline cpp/log/cert_test.cc]: Throughout the test, these `EXPECT_TRUE` should be `EXPECT_OK` instead.

File: cpp/log/cert_test.cc [hunk 2]
```diff
@@ -691,23 +691,23 @@ TEST_F(CertChainTest, RemoveCert) {
 TEST_F(CertChainTest, IssuerChains) {
   // A single certificate.
   CertChain chain(leaf_pem_);
-  EXPECT_EQ(Cert::TRUE, chain.IsValidCaIssuerChainMaybeLegacyRoot());
+  EXPECT_TRUE(chain.IsValidCaIssuerChainMaybeLegacyRoot().ok());
```
Comment (pphaneuf) [inline cpp/log/cert_test.cc]: Would be better to have this as `EXPECT_THAT(..., StatusIs(...));`.

**Comment by pphaneuf** [comment]
LGTM for the main code, test could be tweaked a bit, but I could live with it like this too.

**Comment by Martin2112** [comment]
By "tweaked" do you mean something else than using EXPECT_OK? I'm going to fix all those as it's just regex replaces.

**Comment by Martin2112** [comment]
I've rebased and plan to merge if it passes. If there are any other test changes to be done let me know and I'll follow up.

**Pull request merged**